src/DDTrace/Integrations/Kafka/KafkaIntegration.php

```diff
@@ -0,0 +1,233 @@
+use DDTrace\Integrations\Integration;
+use DDTrace\SpanLink;
+use DDTrace\Tag;
+use DDTrace\Type;
+use DDTrace\Util\ObjectKVStore;
+
+class KafkaIntegration extends Integration
+{
+    const NAME = 'kafka';
+
+    const METADATA_MAPPING = [
+        'metadata.broker.list' => Tag::KAFKA_HOST_LIST,
+        'group.id' => Tag::KAFKA_GROUP_ID,
+        'client.id' => Tag::KAFKA_CLIENT_ID
+    ];
+
+    public function init(): int
+    {
+        if (strpos(phpversion('rdkafka'), '6.') !== 0) {
+            return Integration::NOT_LOADED;
+        }
+
+        $this->installProducerTopicHooks();
+        $this->installConsumerHooks();
+        $this->installConfigurationHooks();
+
+        return Integration::LOADED;
+    }
+
+    private function installProducerTopicHooks()
+    {
+        $integration = $this;
+        \DDTrace\install_hook(
+            'RdKafka\ProducerTopic::producev',
+            function (HookData $hook) use ($integration) {
+                /** @var \RdKafka\ProducerTopic $this */
+                $integration->setupKafkaProduceSpan($hook, $this);
+            }
+        );
+    }
+
+    public function setupKafkaProduceSpan(HookData $hook, \RdKafka\ProducerTopic $producerTopic)
+    {
+        /** @var \RdKafka\ProducerTopic $this */
+        $span = $hook->span();
+        KafkaIntegration::setupCommonSpanMetadata($span, Tag::KAFKA_PRODUCE, Tag::SPAN_KIND_VALUE_PRODUCER, Tag::MQ_OPERATION_SEND);
+
+        $span->meta[Tag::MQ_DESTINATION] = $producerTopic->getName();
+        $span->meta[Tag::MQ_DESTINATION_KIND] = Type::QUEUE;
+
+        $conf = ObjectKVStore::get($producerTopic, 'conf');
+        KafkaIntegration::addProducerSpanMetadata($span, $conf, $hook->args);
+
+        $headers = \DDTrace\generate_distributed_tracing_headers();
+        $hook->args = $this->injectHeadersIntoArgs($hook->args, $headers);
+        $hook->overrideArguments($hook->args);
+    }
+
+    public static function addProducerSpanMetadata($span, $conf, $args)
+    {
+        self::addMetadataToSpan($span, $conf);
+        $span->metrics[Tag::KAFKA_PARTITION] = $args[0];
+        $span->metrics[Tag::MQ_MESSAGE_PAYLOAD_SIZE] = strlen($args[2]);
+        if (isset($args[3])) {
+            $span->meta[Tag::KAFKA_MESSAGE_KEY] = $args[3];
+        }
+    }
+
+    private function injectHeadersIntoArgs(array $args, array $headers): array
+    {
+        // public RdKafka\ProducerTopic::producev (
+        //      integer $partition ,
+        //      integer $msgflags ,
+        //      string $payload [,
+        //      string $key = NULL [,
+        //      array $headers = NULL [,
+        //      integer $timestamp_ms = NULL [,
+        //      string $opaque = NULL
+        // ]]]] ) : void
+        $argsCount = count($args);
+        if ($argsCount >= 5) {
+            $args[4] = array_merge($args[4] ?? [], $headers);
+        } elseif ($argsCount === 4) {
+            $args[] = $headers;
+        } else { // $argsCount === 3
```

Comment:
Please actually add the if () here. Hooks are called before an ArgumentCountError would be emitted.
Just don't do anything if there are not enough args.